services/121-service/src/payments/payments.controller.ts

```diff
@@ -284,22 +285,18 @@ export class PaymentsController {
   @Get('programs/:programId/transactions')
   public async exportTransactionsUsingDateFilter(
     @Res() res: Response,
-    @Param('programId', ParseIntPipe)
-    programId: number,
-    @Query('fromDate') fromDate?: string,
-    @Query('toDate') toDate?: string,
-    @Query('format') format = 'json',
-    @Query('paymentId', new ParseIntPipe({ optional: true }))
-    paymentId?: number,
+    @Param('programId', ParseIntPipe) programId: number,
+    @Query()
+    query: GetTransactionsQueryDto,
   ): Promise<Response | void> {
     const result =
       await this.paymentsReportingService.exportTransactionsUsingDateFilter({
         programId,
-        fromDateString: fromDate,
-        toDateString: toDate,
-        paymentId,
+        fromDateString: query.fromDate,
+        toDateString: query.toDate,
+        paymentId: query.paymentId ? Number(query.paymentId) : undefined,
```

Comment:
Can't/shouldn't we use [NestJS' `ValidationPipe`](https://docs.nestjs.com/techniques/validation) here? As in: we already defined what's valid in the DTO, why do we need to do the casting manually? Or maybe I'm missing something?